assets/src/state.ts

```diff
@@ -1,12 +1,14 @@
 import { Dispatch as ReactDispatch } from "react"
-import { RouteId } from "./skate.d"
+import { RouteId, Vehicle } from "./skate.d"
 
 export interface State {
   selectedRouteIds: RouteId[]
+  selectedVehicle?: Vehicle
```

Comment:
For the vehicle properties panel to be auto-updating, this should be an id, and `app.tsx` should look up the vehicle from its up-to-date list before passing to `<VehicleProperties>`

```suggestion
  selectedVehicleId?: VehicleId
```